nova/virt/vmwareapi/volumeops.py

```diff
@@ -703,21 +703,10 @@ def _detach_volume_fcd(self, connection_info, instance):
             if state != power_state.SHUTDOWN:
                 raise exception.Invalid(_('%s does not support disk '
                                           'hotplug.') % adapter_type)
-        cf = self._session.vim.client.factory
-        fcd_id = data['id']
-        ds_ref_val = data['ds_ref_val']
-        disk_id = vm_util._create_fcd_id_obj(cf, fcd_id)
-        vstorage_mgr = self._session.vim.service_content.vStorageObjectManager
-        fcd_obj = self._session.invoke_api(
-            self._session.vim,
-            'RetrieveVStorageObject',
-            vstorage_mgr,
-            id=disk_id,
-            datastore=ds_ref_val)
-        vmdk_path = fcd_obj.config.backing.filePath
-        hw_devs = vm_util.get_hardware_devices(self._session, vm_ref)
-        device = vm_util.get_vmdk_volume_disk(hw_devs, vmdk_path)
-        volume_uuid = fcd_obj.config.name.replace('volume-', '')
+        volume_uuid = connection_info['volume_id']
```

Comment:
https://github.com/sapcc/nova/commit/bec171b92b40a7564c85e468dc29ef833191d973
we might want to handle that, too, I guess

Reply:
The mentioned function is returning the serial as failback, i'm not sure that is good or not, i need to check the codes

Reply:
https://github.com/sapcc/nova/blob/stable/xena-m3/nova/virt/block_device.py#L571-L575
It looks like, volume_id is always populated, and serial is only when no serial is returned, i checked fcd does not return serial, but i don't see a benefit to use the mentioned function and not directly grab the volume_id from the connection_info ....

Reply:
hm ... the point of the commit was, that `volume_id` might not always be in `connection_info`. not sure when exactly that happens, but it seems to have happened before.

Since there's a helper to get the volume-id from the connection_info and we need the volume-id, I'd think it would be best to use it wherever we can.

But if you think that has negative effects, we can fix it when it hits us .